examples/hosts/app-integration/external-data/src/model/taskList.ts

```diff
@@ -43,6 +52,44 @@ class Task extends TypedEventEmitter<ITaskEvents> implements ITask {
         this._priority.on("valueChanged", () => {
             this.emit("priorityChanged");
         });
+        this.diffType = "none";
+        this.diffName = "";
+        this.diffPriority = Number.POSITIVE_INFINITY;
+    }
+    public externalNameChanged = async (savedName: string): Promise<void> => {
+        this.diffName = savedName;
+        this.diffType = "change";
+        this.emit('externalNameChanged');
+    }
+    public externalPriorityChanged = async (savedPriority: number): Promise<void> => {
+        this.diffPriority = savedPriority;
+        this.diffType = "change";
+        this.emit('externalPriorityChanged');
+    }
+    public acceptChange = async (): Promise<void> => {
+        if (this.diffPriority !== this.DEFAULT_PRIORITY) {
+            this._priority.set(this.diffPriority);
+        }
+        if (this.diffName !== '') {
+            const oldString = this._name.getText()
+            console.log(this.name.getText());
+            this._name.replaceText(0, oldString.length, this.diffName);
+            console.log(this.name.getText());
+        }
+        this.diffPriority = this.DEFAULT_PRIORITY;
+        this.diffType = "none";
+        this.emit('externalPriorityChanged');
+        this.diffName = '';
+        this.diffType = "none";
+        this.emit('externalNameChanged');
+    }
+    public ignoreChange = async (): Promise<void> => {
+        this.diffPriority = this.DEFAULT_PRIORITY;
+        this.diffType = "none";
+        this.emit('externalPriorityChanged');
+        this.diffName = '';
+        this.diffType = "none";
+        this.emit('externalNameChanged');
```

Comment:
Emitting here feels weird. If we're ignoring the external changes I don't think these should happen. If anything, they should emit when we get the notification (or poll for) that the external data source had changes?

Reply:
These emits are to trigger re-rending of the React component.

Reply:
Moved to getter setter pattern and that definitely cleans up this mess of triggers!